### ISSUE TITLE
Headers attribute refers to cells in same table -  problems with examples

### ISSUE THREAD

**Issue by dd8** [act-rules/act-rules.github.io#1195]
# Inapplicable Example 3 causes a fail of 1.3.1:
https://act-rules.github.io/rules/a25f45#inapplicable-example-3

Although the table is hidden offscreen it is voiced by screen readers (unless the browser used is Firefox, which treats offscreen content differently to Chrome/Safari/IE11). It's also a fail of SC 1.3.1 for non-screen reader users since the table is only available to screen reader users. From Understanding SC 1.3.1:

> When such relationships are perceivable to one set of users, those relationships can be made to be perceivable to all. One method of determining whether or not information has been properly provided to all users is to access the information serially in different modalities.

https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html

# Passed Example 3 doesn't work in most AT:
VoiceOver - the `td` referenced by `headers` isn't read as a header
JAWS 2019 with IE11 - table not read at all (complete silence)
JAWS 2019 with FF - table not read at all (complete silence)
JAWS 2019 with Chrome - read correctly with table header
NVDA 2019 with IE11 - the `td` referenced by `headers` isn't read as a header
NVDA 2019 with FF - the `td` referenced by `headers` isn't read as a header
NVDA 2019 with Chrome - the `td` referenced by `headers` isn't read as a header

# 25% of the examples are treated as layout tables
About 25% of the example tables are treated as layout tables by NVDA and JAWS heuristics because they contain no `th` elements and have few rows and columns:
https://act-rules.github.io/rules/a25f45#failed-example-3
https://act-rules.github.io/rules/a25f45#inapplicable-example-2
https://act-rules.github.io/rules/a25f45#inapplicable-example-3
https://act-rules.github.io/rules/a25f45#inapplicable-example-5

**Comment by JAWS-test** [comment]
> Although the table is hidden offscreen it is voiced by screen readers (unless the browser used is Firefox, which treats offscreen content differently to Chrome/Safari/IE11). 

I checked the (offscreen table) [https://act-rules.github.io/rules/a25f45#inapplicable-example-3] in Firefox with JAWS and NVDA: It is output by both screen readers
* JAWS 2019: as text
* NVDA 2019: as table

**Comment by JAWS-test** [comment]
I used an extended [example](https://codepen.io/jaws-test/pen/GRJOYRg) for testing of "Passed Example 3" to have 2 columns (for testing table navigation with Ctrl+Alt+arrow keys) and get very different test results than @dd8:

* first table: single column
* second table: 2 columns

JAWS 2019
* IE 11 and Chrome 80: both as table, correct output of table headers
* Firefox 73.0.1: first as text, second as table (correct output of table headers)

NVDA 2019.2.1:
* IE 11: both as table, no output of the columnheaders during navigation through table cells
* Chrome: both as table, correct output of table headers
* Firefox: both as table, correct output of table headers

**Comment by jkodu** [comment]
@dd8 I will go through your test results in detail. But here is a wiki with my test findings - https://github.com/act-rules/act-rules.github.io/wiki/Assistive-technology-tests-on-table-header-associations

**Comment by Jym77** [comment]
Note that having inapplicable/passed examples that fail the SC is **not** a problem per se. Most rules are in that case and this is the reason why the accessibility mapping usually says "passed: further test is needed".

Of course, we should try and expand rules to catch these when it's easy to do so. But it is perfectly OK to have rules that are stricter than the SC.

**Comment by dd8** [comment]
> I used an extended [example](https://codepen.io/jaws-test/pen/GRJOYRg) for testing of "Passed Example 3" to have 2 columns (for testing table navigation with Ctrl+Alt+arrow keys) and get very different test results than @dd8:
> 
Changing the examples affects whether browsers treat the tables as data tables or layout tables. There's a summary of the  algorithms browsers / AT use here:
https://www.powermapper.com/blog/layout-tables-vs-data-tables/

JAWS in particular has a very weird algorithm, which is affected by default font size and screen size when table width is relative to viewport (e.g. width: 100%)
https://doccenter.freedomscientific.com/doccenter/doccenter/rs11f929e9c511/2012-05-08_teachersandtrainers-l11/02_tablesandforms.htm

In JAWS 2019 with IE11 or FF you can toggle a width:100% table with no `th` headers between layout (table not announced) and data just by resizing the window and hitting refresh

```
<table style="width:100%">
<tr>
<td>At small window sizes</td>
<td>I am data</td>
</tr>
<tr>
<td>At large window sizes</td>
<td>I am layout</td>
</tr>
</table>
```
